src/services/codefixes/generateAccessors.ts

```diff
@@ -251,6 +251,20 @@ namespace ts.codefix {
         });
     }
 
+    function getDeclarationType(declaration: AcceptedDeclaration, program: Program): TypeNode | undefined {
+        const typeNode = getTypeAnnotationNode(declaration);
+        const strictNullChecks = getStrictOptionValue(program.getCompilerOptions(), "strictNullChecks");
+        if (isPropertyDeclaration(declaration) && typeNode && declaration.questionToken && strictNullChecks) {
+            const typeChecker = program.getTypeChecker();
+            const type = typeChecker.getTypeFromTypeNode(typeNode);
+            if (!(type.flags & TypeFlags.AnyOrUnknown || typeChecker.getFalsyFlags(type) & TypeFlags.Undefined)) {

```

Comment:
`getFalsyFlags()` is a really weird function, and it seems like you only need it for collecting the flags over possible unions... a couple options for doing the same thing without exporting this function:

- `typeChecker.getUnionType([type, typeChecker.getUndefinedType()], UnionReduction.Subtype) === type`
- `typeChecker.isTypeAssignableTo(typeChecker.getUndefinedType(), type)`
- Write a utility that combines _all_ the flags over union types, not the unintuitive mapping that `getFalsyFlags` does (I’m kind of surprised that doesn’t exist)

The first two are less efficient than what you have, but I think they’d be totally acceptable—IMO it’s worth it to keep the use of `getFalsyFlags` very limited.

Reply:
@andrewbranch Thanks for the detailed feedback 👍🏻.